atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java

```diff
@@ -775,6 +779,51 @@ public void deletesGetBatched() {
                 .isEqualTo(maxTsForFinePartition(0));
     }
 
+    @Test
+    public void multipleSweepersSweepDifferentShardsAndCallUnlockAfterwards() throws InterruptedException {
+        int shards = 128;
+        int sweepers = 8;
+        int threads = shards / sweepers;
+        LockService stickyLockService = createStickyLockService();
+        createAndInitializeSweepersAndWaitForOneBackgroundIteration(sweepers, shards, threads, stickyLockService);
+
+        for (int i = 0; i < shards; i++) {
+            assertProgressUpdatedToTimestamp(maxTsForFinePartition(tsPartitionFine(unreadableTs - 1)), i);
+            verify(stickyLockService, times(1)).unlock(new LockRefreshToken(BigInteger.valueOf(i), 1L));
+        }
+    }
+
+    @Test
+    public void extraSweepersGiveUpAfterFailingToAcquireEnoughTimes() throws InterruptedException {
+        int shards = 8;
+        int sweepers = 2;
+        int threads = shards / sweepers + 1;
+        LockService stickyLockService = createStickyLockService();
+        createAndInitializeSweepersAndWaitForOneBackgroundIteration(sweepers, shards, threads, stickyLockService);
+
+        // We issue between 2 * shards - shards/sweepers and 2 * shards - 1 lock requests to acquire all locks, and then
+        // an additional 2 * shards requests by the unlucky threads that have been unable to get a lock
+        ArgumentCaptor<LockRequest> captor = ArgumentCaptor.forClass(LockRequest.class);
+        verify(stickyLockService, atLeast(2 * shards - shards / sweepers + 2 * shards))
```

Comment:
May be worth extracting something or a comment for why there's the hanging `2 * shards` here